extension/android/jni/jni_layer.cpp

```diff
@@ -508,6 +508,12 @@ class ExecuTorchJni : public facebook::jni::HybridClass<ExecuTorchJni> {
 };
 } // namespace executorch::extension
 
+#ifdef EXECUTORCH_BUILD_ASR_JNI
+extern void register_natives_for_asr();
+#else
+void register_natives_for_asr() {}
+#endif
+
```

Comment:
Semi related to the diff, I think we started to recognize a plugin pattern here, it would be better if we can rely on static initialization to register these natives and just iterate through them in line 535. This way we don't need all these macros, we just link the necessary library and it register itself.